pkg/mcp/server/server.go

```diff
@@ -465,7 +465,13 @@ func (con *connection) receive() {
 			con.reqError = err
 			return
 		}
-		con.requestC <- req
+		select {
+		case <-con.stream.Context().Done():
+			scope.Debugf("MCP: connection %v: stream done, err=%v", con, con.stream.Context().Err())
+			return
+		default:
+			con.requestC <- req
+		}
```

Comment:
There is still a race between the stream being canceled and posting to the channel. The https://github.com/istio/istio/pull/1086 equivalent for MCP would be something like the following:

```golang
case <-con.stream.Context().Done():
	scope.Debugf("MCP: connection %v: stream done, err=%v", con, con.stream.Context().Err())
	return
case con.requestC <- req:
}
```